exporter/exporterhelper/internal/persistent_queue.go

```diff
@@ -28,50 +27,35 @@ var (
 // persistentQueue holds the queue backed by file storage
 type persistentQueue struct {
 	*persistentContiguousStorage
-	stopWG       sync.WaitGroup
-	set          exporter.CreateSettings
-	storageID    component.ID
-	numConsumers int
+	set       exporter.CreateSettings
+	storageID component.ID
+	dataType  component.DataType
 }
 
 // NewPersistentQueue creates a new queue backed by file storage; name and signal must be a unique combination that identifies the queue storage
-func NewPersistentQueue(capacity int, numConsumers int, storageID component.ID, marshaler QueueRequestMarshaler,
-	unmarshaler QueueRequestUnmarshaler, set exporter.CreateSettings) Queue {
+func NewPersistentQueue(capacity int, dataType component.DataType, storageID component.ID,
+	marshaler QueueRequestMarshaler, unmarshaler QueueRequestUnmarshaler, set exporter.CreateSettings) Queue {
 	return &persistentQueue{
 		persistentContiguousStorage: newPersistentContiguousStorage(set.Logger, uint64(capacity), marshaler, unmarshaler),
-		numConsumers:                numConsumers,
 		set:                         set,
 		storageID:                   storageID,
+		dataType:                    dataType,
 	}
 }
 
 // Start starts the persistentQueue with the given number of consumers.
-func (pq *persistentQueue) Start(ctx context.Context, host component.Host, set QueueSettings) error {
-	storageClient, err := toStorageClient(ctx, pq.storageID, host, pq.set.ID, set.DataType)
+func (pq *persistentQueue) Start(ctx context.Context, host component.Host) error {
+	storageClient, err := toStorageClient(ctx, pq.storageID, host, pq.set.ID, pq.dataType)
 	if err != nil {
 		return err
 	}
 	pq.persistentContiguousStorage.start(ctx, storageClient)
-	for i := 0; i < pq.numConsumers; i++ {
-		pq.stopWG.Add(1)
-		go func() {
-			defer pq.stopWG.Done()
-			for {
-				req, found := pq.persistentContiguousStorage.get()
-				if !found {
-					return
-				}
-				set.Callback(req)
-			}
-		}()
-	}
 	return nil
 }
 
 // Shutdown stops accepting items, shuts down the queue and closes the persistent queue
 func (pq *persistentQueue) Shutdown(ctx context.Context) error {
 	close(pq.persistentContiguousStorage.stopChan)
-	pq.stopWG.Wait()
 	return stopStorage(pq.persistentContiguousStorage.client, ctx)
```

Comment:
In order for this to work, you need https://github.com/open-telemetry/opentelemetry-collector/pull/8899